File: app/src/main/java/com/example/myvotingapp/loginscreen.java
package com.example.myvotingapp;

import android.app.Activity;

public class loginscreen extends Activity {
}


@Composable
fun LoginScreen(navController: NavHostController, authViewModel: AuthViewModel = androidx.lifecycle.viewmodel.compose.viewModel()) {
var id by remember { mutableStateOf("") }
var password by remember { mutableStateOf("") }
val loginState by authViewModel.loginState.collectAsState()

Column(
        modifier = Modifier
        .fillMaxSize()
            .padding(16.dp)
    ) {
TextField(value = id, onValueChange = { id = it }, label = { Text("South African ID") })
TextField(value = password, onValueChange = { password = it }, label = { Text("Password") }, visualTransformation = PasswordVisualTransformation())
Spacer(modifier = Modifier.height(16.dp))
Button(onClick = { authViewModel.login(id, password) }) {
Text("Log In")
        }
Spacer(modifier = Modifier.height(8.dp))
Button(onClick = { navController.navigate("signup") }) {
Text("Sign Up")
        }
                }

                if (loginState.success) {
        navController.navigate("dashboard")
    }
            }

File: app/src/main/java/com/example/myvotingapp/loginscreen.kt
@file:OptIn(ExperimentalMaterial3Api::class)

package com.example.myvotingapp

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.navigation.NavHostController
import com.example.myvotingapp.myvotingapp.AuthViewModel

@Composable
fun LoginScreen(navController: NavHostController, authViewModel: AuthViewModel = androidx.lifecycle.viewmodel.compose.viewModel()) {
    var id by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    val loginState by authViewModel.loginState.collectAsState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp)
    ) {
        TextField(value = id, onValueChange = { id = it }, label = { Text("South African ID") })
        TextField(value = password, onValueChange = { password = it }, label = { Text("Password") }, visualTransformation = PasswordVisualTransformation())
        Spacer(modifier = Modifier.height(16.dp))
        Button(onClick = { authViewModel.login(id, password) }) {
            Text("Log In")
        }
        Spacer(modifier = Modifier.height(8.dp))
        Button(onClick = { navController.navigate("signup") }) {
            Text("Sign Up")
        }
    }

    if (loginState.success) {
        navController.navigate("dashboard")
    }
}

